### ISSUE TITLE
Sport Live errors in almost all sections

### ISSUE THREAD

**Issue by nicoali** [mbebe/blomqvist#812]
In almost all sections and links happens this
![image](https://user-images.githubusercontent.com/14332128/181291188-2aba8754-b685-49d6-afa9-22252756515a.png)

**Comment by nicoali** [comment]
here it is a log that i found:
https://pastebin.com/Az8kBiPf

**Comment by mtr81** [comment]
Hi 
Pls. check that version
[plugin.video.sportliveevents.zip](https://github.com/mbebe/blomqvist/files/9204183/plugin.video.sportliveevents.zip)
Daddylive and klubsports streams should be working (changes in daddylive links resolvers).
As for stream2watch - the structure of the www has changed, more time is needed to fix this part of the add-ons

**Comment by nicoali** [comment]
Thx i'll test it!

why did you removed vipleague?

**Comment by mtr81** [comment]
Sorry, I was working on old version. Below fix v1.9:
[plugin.video.sportliveevents.zip](https://github.com/mbebe/blomqvist/files/9209706/plugin.video.sportliveevents.zip)
BTW. streams2watch still remains to be fix.

**Comment by akeotaseo** [comment]
vipleague doesn't work?...

**Comment by akeotaseo** [comment]
> Sorry, I was working on old version. Below fix v1.9: [plugin.video.sportliveevents.zip](https://github.com/mbebe/blomqvist/files/9209706/plugin.video.sportliveevents.zip) BTW. streams2watch still remains to be fix.

vipleague doesn't work?...

**Comment by mtr81** [comment]
I didn't check vipleague - only daddylive and klubsports.

**Comment by nicoali** [comment]
daddylive and klubsport are ok. Tested. While vipleague and others are still off

**Comment by nicoali** [comment]
Also livesporthd.live section that few days ago worked fine now is out

**Comment by mtr81** [comment]
What exactly in SportHD.live section isn't working? I check some live TV and links in events and everything works fine...

**Comment by nicoali** [comment]
I cannot enter in channels and schedule section with Sport LIVE error check log. Later i'll post the log

**Comment by nicoali** [comment]
with Kodi 19.4 on Personal Computer SportHD.live works fine
while on Kodi 19.3 on amazon fire stick i have Sport LIVE error check log i'm tryng to find logs! do you know where are stored on amazon firestick?

**Comment by nicoali** [comment]
i've installed kodi 19.4 on amazon firestick to have same conditions:

same error of before..

here it is the log
https://pastebin.com/JNPtVsTQ

**Comment by zkte** [comment]
@nicoali it is blocked by your ISP or DNS.

**Comment by nicoali** [comment]
@zkte thxs i changed dns on firestick and now the section SportHD.live is fine...

What about vipleague??

**Comment by zkte** [comment]
vipleague blocks the TLS signature of kodi ffmpeg/curl, need to use a proxy that uses same parameters as a real browser.

**Comment by nicoali** [comment]
this is the log:
https://pastebin.com/w1JT60XL

how to do that?  £use proxy that uses same TLS parameters as a real browser."

we need a fix of mbebe or i can do something?

**Comment by zkte** [comment]
you could do something using stunnel and transparent proxy but it's complicated.

**Comment by nicoali** [comment]
@zkte 
mbebe can do some mod that can fix this?

**Comment by mtr81** [comment]
I implemented gettiing the HLS key via Python's internal proxy (requests with user-agent and referer headers same as browser), but that didn't fix the problem.

**Comment by nicoali** [comment]
@mtr81 thx for your effort.

Maybe they changed something in the website because 10 days ago the version of mbebe was working

**Comment by mtr81** [comment]
HLS Manifest is something like this:
```
#EXTM3U
#EXT-X-VERSION:4
#EXT-X-TARGETDURATION:4
#EXT-X-MEDIA-SEQUENCE:4043
#EXT-X-DISCONTINUITY-SEQUENCE:0
#EXT-X-KEY:METHOD=AES-128,URI="https://key.seckeyserv.me/beviziyiluv0xo3a2o4o/1659102039",IV=0xd0e661294f14418bf80ccaa46c8d979a,KEYFORMAT="identity",KEYFORMATVERSIONS="1"
#EXTINF:4.0,
media_4043.ts
#EXTINF:4.0,
media_4044.ts
#EXTINF:4.0,
media_4045.ts
#EXTINF:4.0,
media_4046.ts
#EXTINF:4.0,
media_4047.ts
#EXTINF:4.0,
media_4048.ts
#EXTINF:4.0,
media_4049.ts
#EXTINF:4.0,
media_4050.ts
```
The problem is getting response from 'key.seckeyserv.me'.  I used in internal proxy the same headers as a browser but without success.

**Comment by zkte** [comment]
@mtr81 like i said you need correct TLS parameters the following works:
```
curl 'https://key.seckeyserv.me/xa3uhip0duseme4ebeyi/1659100985' \
-H 'origin: https://www.liveply.me' \
-H 'referer: https://www.liveply.me/sd0embed?v=ky2hd~ky2sd' \
-H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="99", "Google Chrome";v="99"' \
-H 'sec-ch-ua-mobile: ?1' \
-H 'sec-ch-ua-platform: "Android"' \
-H 'Upgrade-Insecure-Requests: 1' \
-H 'User-Agent: Mozilla/5.0 (Linux; Android 12; Pixel 6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.58 Mobile Safari/537.36' \
-H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9' \
-H 'Sec-Fetch-Site: none' \
-H 'Sec-Fetch-Mode: navigate' \
-H 'Sec-Fetch-User: ?1' \
-H 'Sec-Fetch-Dest: document' \
-H 'Accept-Encoding: gzip, deflate, br' \
-H 'Accept-Language: en-US,en;q=0.9' \
--curves "X25519:P-256:P-384" \
--ciphers TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_CHACHA20_POLY1305_SHA256,ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-RSA-AES128-GCM-SHA256,ECDHE-ECDSA-AES256-GCM-SHA384,ECDHE-RSA-AES256-GCM-SHA384,ECDHE-ECDSA-CHACHA20-POLY1305,ECDHE-RSA-CHACHA20-POLY1305,ECDHE-RSA-AES128-SHA,ECDHE-RSA-AES256-SHA,AES128-GCM-SHA256,AES256-GCM-SHA384,AES128-SHA,AES256-SHA \
--http2 --false-start --compressed
```

**Comment by Dilan12891** [comment]
Daddylive and vipleague arent working for me.
Hope vipleagu Will fix bc I use it a lot

**Comment by zkte** [comment]
with stunnel you can use conf
```
foreground = yes
[key.seckeyserv.me]
client = yes
accept = localhost:8443
connect = key.seckeyserv.me:443
sni = key.seckeyserv.me
OCSPaia = no
curves = X25519:P-256:P-384
```
then proxy the key with Host header
```
curl 'http://localhost:8443/xa3uhip0duseme4ebeyi/1659100985' \
-H 'origin: https://www.liveply.me' \
-H 'referer: https://www.liveply.me/sd0embed?v=ky2hd~ky2sd' \
-H 'User-Agent: Mozilla/5.0 (Linux; Android 12; Pixel 6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.58 Mobile Safari/537.36' \
-H 'Accept: */*' \                                                                                                                                    
-H 'Accept-Encoding: gzip, deflate, br' \
-H 'Accept-Language: en-US,en;q=0.9' \
-H 'Host: key.seckeyserv.me'
```
in Python3 there is [ssl.SSLContext.set_ecdh_curve()](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.set_ecdh_curve) method. it's not fully implemented but could work using a requests adapter.
```
from requests.adapters import HTTPAdapter
from requests.packages.urllib3.util.ssl_ import create_urllib3_context

class KeyAdapter(HTTPAdapter):
    def init_poolmanager(self, *args, **kwargs):
        context = create_urllib3_context(ciphers="DEFAULT:!DHE:!SHA1:!SHA256:!SHA384")
        context.set_ecdh_curve("prime256v1")
        kwargs["ssl_context"] = context
        return super(KeyAdapter, self).init_poolmanager(*args, **kwargs)

sess = requests.Session()
sess.mount("https://key.seckeyserv.me/", KeyAdapter())
```

**Comment by nicoali** [comment]
Is it possible to have an how to do.. For dummies or build a zip ready to
use??

Or a tutorial??

Thx in advance



Il dom 31 lug 2022, 07:54 zkte ***@***.***> ha scritto:

**Comment by mtr81** [comment]
@nicoali check this version:
[plugin.video.sportliveevents.zip](https://github.com/mbebe/blomqvist/files/9228916/plugin.video.sportliveevents.zip)
I tested streams from section Vipleauge - > LIVE NOW and most of them are working (some problems with WTA Prague).

@zkte thanks for your help

**Comment by nicoali** [comment]
Thanks now i test it.

Is there a new issue for the section kubsports??
Links are missing

Il dom 31 lug 2022, 15:12 mtr81 ***@***.***> ha scritto:

**Comment by mtr81** [comment]
I check links in vipleague and I don't see any problems... Which transmission doesn't work?
Klubsports- probably changes in www structures.

**Comment by nicoali** [comment]
For vipleague i can see links... Seems that start to load.. Then i have
playback failed

One or more items failed to play.

I attached also log in the other post.

Almost all links live of football

Il dom 31 lug 2022, 19:04 mtr81 ***@***.***> ha scritto:

**Comment by mtr81** [comment]
Can you paste full debug log?

**Comment by nicoali** [comment]
@mtr81 

here it is log:
https://paste.kodi.tv/ozosexuxeg.kodi

**Comment by mtr81** [comment]
This is not debug log (settings->system->logging->enable debug logging (ON) and enable event logging (ON))

**Comment by Dilan12891** [comment]
For me it works fine

**Comment by nicoali** [comment]
@mtr81 
https://paste.kodi.tv/cowimenamu.kodi

hope this is good!

@Dilan12891 lucky man!

**Comment by zkte** [comment]
don't think you need to proxy the .ts segments, unless it is something windows-specific. then maybe try with inputstream.adaptive

**Comment by mbebe** [comment]
@nicoali are You sure, you have installed full zip ?
It looks like you still have previous version of main.py file.
In this version main.py is playing vipleague stream via INPUTSTREAMadaptive.

**Comment by nicoali** [comment]
Yes i have full uninstalles the plugin and installed zipfile posted from mtr81.
I will uninstall the plugin clear cache and reinstall all

**Comment by mtr81** [comment]
[plugin.video.sportliveevents-1.9a.zip](https://github.com/mbebe/blomqvist/files/9229672/plugin.video.sportliveevents-1.9a.zip)
Install above version + paste log (I added some information to print in log)

**Comment by nicoali** [comment]
i've installed 1.9a from zip

here log:
https://paste.kodi.tv/zaxalucawe.kodi

**Comment by mtr81** [comment]
In your log I do not see one important thing:
`INFO <general>: CAddonMgr::ADDON::CAddonMgr::FindAddons: inputstream.adaptive v19.0.7 installed`
Pls. check Settings->Add-ons->VideoPlayer Inputstream:
![isa](https://user-images.githubusercontent.com/86961563/182052270-d5f130e4-73ee-4f47-be0d-25bd63a36e10.png)
How @mbebe wrote, it seems that ISA doesn't start.
Anyway, I paste to check cleanned version of sportlive addon.
[plugin.video.sportliveevents-1.10.zip](https://github.com/mbebe/blomqvist/files/9230080/plugin.video.sportliveevents-1.10.zip)

**Comment by nicoali** [comment]
I have reinstalled 
![182052270-d5f130e4-73ee-4f47-be0d-25bd63a36e10](https://user-images.githubusercontent.com/14332128/182068355-315730f6-150d-4603-8880-6f92152e1df8.png)
And seems all fine now

Thx @mtr81 @zkte @mbebe

**Comment by mtr81** [comment]
Fix klubsports i partially streams2watch
[plugin.video.sportliveevents-1.10.zip](https://github.com/mbebe/blomqvist/files/9238159/plugin.video.sportliveevents-1.10.zip)

**Comment by zkte** [comment]
daddylive says closed, maxsport.one is alternative

**Comment by mtr81** [comment]
@zkte 
Check _livetvon.click_

**Comment by akeotaseo** [comment]
> daddylive

https://daddylive.futbol/

**Comment by mtr81** [comment]
Fix daddylive and s2w sections
[plugin.video.sportliveevents-1.11.zip](https://github.com/mbebe/blomqvist/files/9300549/plugin.video.sportliveevents-1.11.zip)

**Comment by nicoali** [comment]
> Fix daddylive and s2w sections [plugin.video.sportliveevents-1.11.zip](https://github.com/mbebe/blomqvist/files/9300549/plugin.video.sportliveevents-1.11.zip)

daddylive works fine.. there is only an issue in schedule section! the date is wrong! 8 of August... instead of today 10

**Comment by mtr81** [comment]
Check daddylive.futbol... I can't see schedule on 10th August...

**Comment by zkte** [comment]
livetvon.click and daddylive.futbol are dead

**Comment by mtr81** [comment]
maxsport.one ->
[plugin.video.maxsport.zip](https://github.com/mbebe/blomqvist/files/9331657/plugin.video.maxsport.zip)

**Comment by nicoali** [comment]
Today

> > Fix daddylive and s2w sections [plugin.video.sportliveevents-1.11.zip](https://github.com/mbebe/blomqvist/files/9300549/plugin.video.sportliveevents-1.11.zip)


From today i have problem with all links of vipleague. Do you have same issue??

**Comment by mtr81** [comment]
I confirm

**Comment by mtr81** [comment]
It seems I fixed vipleague, but now playback starts very slowly...
[plugin.video.sportliveevents-1.12.zip](https://github.com/mbebe/blomqvist/files/9333436/plugin.video.sportliveevents-1.12.zip)

**Comment by nicoali** [comment]
> It seems I fixed vipleague, but now playback starts very slowly... [plugin.video.sportliveevents-1.12.zip](https://github.com/mbebe/blomqvist/files/9333436/plugin.video.sportliveevents-1.12.zip)

yes works but playback starts on firestick in more than 2 minutes

**Comment by mtr81** [comment]
Check that version ->
[plugin.video.sportliveevents-1.12.zip](https://github.com/mbebe/blomqvist/files/9333981/plugin.video.sportliveevents-1.12.zip)
It should be much better...

**Comment by nicoali** [comment]
With version 1.12 i have playback error

**Comment by mtr81** [comment]
Yes. I see that name of variable with stream url now is also... variable (fe. vidPlayUrl, videoUrl). So another code modification:
[plugin.video.sportliveevents-1.12.zip](https://github.com/mbebe/blomqvist/files/9339481/plugin.video.sportliveevents-1.12.zip)

**Comment by nicoali** [comment]
> Yes. I see that name of variable with stream url now is also... variable (fe. vidPlayUrl, videoUrl). So another code modification: [plugin.video.sportliveevents-1.12.zip](https://github.com/mbebe/blomqvist/files/9339481/plugin.video.sportliveevents-1.12.zip)

also with last version:
https://paste.kodi.tv/pokinonoco.kodi

**Comment by zkte** [comment]
vipbox snitches in here lol, dont post your addons here if you dont want to fix it every day.

**Comment by nicoali** [comment]
if someone can fix it can send me to my email address ??
***@***.***

Il giorno mar 16 ago 2022 alle ore 15:45 zkte ***@***.***> ha
scritto:

**Comment by mtr81** [comment]
@zkte  I don't use this addon at all :-)
@nicoali Don't count on me. It's a waste of time.

**Comment by nicoali** [comment]
@mtr81 ok don't worry!

i was looking at code.. the problem is always here
https://paste.kodi.tv/facasaxalu

if you can chat to explain me better in pvt what i have to search in the source code of the webpage  to fix the problem will be great. So i'll not bore you anymore

**Comment by mtr81** [comment]
This time - in that part of code - is another js script to solve (not direct or base64 encoded url stream)

**Comment by Dilan12891** [comment]
Does the vipbox works?

**Comment by mbebe** [comment]
vipbox, vipleague are offline.

**Comment by Dilan12891** [comment]
Could it be fixed pls?

**Comment by leathakkor** [comment]
@zkte:  are any of these methods still working for you to access https://key.seckeyserv.me/ ? I can't seem to figure out what is going on but about 5 days ago most of my machines stopped working with this site, but 2 or 3 kept working and i can't figure out what is going on?

**Comment by zkte** [comment]
@leathakkor I don't see them using AES HLS at the moment, so you can play the stream just normally
```
streamlink --http-header "Referer=https://www.liveply.me/" https://ed-t1.icetoot.me/plyvivo/104u3u30za8odi3oxo4a/chunklist.m3u8 best
```

**Comment by zkte** [comment]
@leathakkor I see they are doing a periodic poll to authorize the IP before you can stream, so you might need to open the stream once before in browser.
```
const formAuthUrl = function(data) {
    if (typeof data.scode === 'undefined' || typeof data.ts === 'undefined') {
        return '';
    }
    return window.atob(authCallUrl) + '?stream=' + unqiueId + '&scode=' + data.scode + '&expires=' + data.ts;
};
const forceKillplayer = function() {
    authUrl = '';
    playerObj.stop();
    playerObj.remove();
    clearTimeout(deauthcron);
    ajxRequest.abort();
    if (playerObj.destroy) {
        playerObj.destroy();
    }
    if (doRedirect) {
        window.location = 'techissue.html';
    }
};
let deauthcron;
const enablePeer = 1;
const enablePeerCache = 1;
const checkPeer = p2pml.hlsjs.Engine.isSupported();
const doRedirect = 1;
const doAuthCheck = function() {
    if (checkPeer && authUrl !== '') {
        ajxRequest = $.ajax({
            url: authUrl,
            dataType: 'json',
            timeout: 5000,
            success: function(response) {
                authUrl = formAuthUrl(response);
                deauthcron = setTimeout(doAuthCheck, 300000);
            },
            error: function() {
                if (failCount < 2) {
                    deauthcron = setTimeout(doAuthCheck, 5000);
                    failCount++;
                } else {
                    forceKillplayer();
                }
            },
            statusCode: {
                444: function() {
                    forceKillplayer();
                },
                410: function() {
                    forceKillplayer();
                },
                403: function() {
                    forceKillplayer();
                }
            }
        });
    }
};
const authCallUrl = 'aHR0cHM6Ly9rZXkuc2Vja2V5c2Vydi5tZQ==';
const playerId = 'g1h7m5x3k9';
const unqiueId = 'kiwu6a9uvi5ujubuxi7o';
const videoUrl = 'aHR0cHM6Ly9lZC10MS5pY2V0b290Lm1lL3BseXZpdm8va2l3dTZhOXV2aTV1anVidXhpN28vY2h1bmtsaXN0Lm0zdTg=';
const kelVal = 'z2f1z1c4z9r4o2';
const jwbasepath = 'https://cdn.liveply.me/scripts/player/8.25.8/';
let failCount = 1;
let ajxRequest;
let authUrl = formAuthUrl({
    "scode": "MpDf4Kib2AColxIhtg-NSg",
    "ts": 1662142892
});
doAuthCheck();
const ratio = $(window).width() / $(window).height();
const aspectRatio = ratio < 1.5 ? '4:3' : '16:9';
let sWarmEngine = new p2pml.hlsjs.Engine({
    segments: {
        swarmId: unqiueId
    },
    loader: {
        trackerAnnounce: ['wss://fast.swarm.video'],
        httpUseRanges: true,
        WaitForTracker: enablePeerCache,
        WaitForTrackerCounter: 1500
    }
});
const playerObj = jwplayer(playerId);
playerObj.setup({
    file: window.atob(videoUrl),
    controls: true,
    autostart: false,
    mute: false,
    hlshtml: true,
    image: 'https://cdn.liveply.me/images/thumb/kiwu6a9uvi5ujubuxi7o.jpeg',
    responsive: true,
    renderCaptionsNatively: false,
    liveSyncDuration: 30,
    aspectratio: aspectRatio,
    width: '100%',
    type: 'hls',
    androidhls: true,
    loadAndParseHlsMetadata: false,
    key: kelVal,
    primary: 'html5',
    preload: 'none',
    hlsjsdefault: true,
    cast: {},
    base: jwbasepath
});
playerObj.on('error', forceKillplayer);
let debounce;
playerObj.on('buffer', function() {
    debounce = setTimeout(forceKillplayer, 15000);
});
playerObj.on('play', function() {
    clearTimeout(debounce);
});
jwplayer_hls_provider.attach();
p2pml.hlsjs.initJwPlayer(playerObj, {
    liveSyncDurationCount: 7,
    loader: sWarmEngine.createLoaderClass()
});
setTimeout(forceKillplayer, 2 * 3600000);
const fakelog = function() {};
window['console']['log'] = fakelog;
setInterval(function() {
    let startTime = performance.now();
    debugger ;let stopTime = performance.now();
    if ((stopTime - startTime) > 100) {
        window.location = 'https://www.youtube.com/watch?v=rqX65RHTWrA'
    }
}, 100);
!function() {
    function detectDevTool(allow) {
        if (isNaN(+allow))
            allow = 100;
        let start = +new Date();
        debugger ;let end = +new Date();
        if (isNaN(start) || isNaN(end) || end - start > allow) {
            window.location = 'https://www.youtube.com/watch?v=rqX65RHTWrA'
        }
    }
    if (window.attachEvent) {
        if (document.readyState === "complete" || document.readyState === "interactive") {
            detectDevTool();
            window.attachEvent('onresize', detectDevTool);
            window.attachEvent('onmousemove', detectDevTool);
            window.attachEvent('onfocus', detectDevTool);
            window.attachEvent('onblur', detectDevTool);
        } else {
            setTimeout(argument.callee, 0);
        }
    } else {
        window.addEventListener('load', detectDevTool);
        window.addEventListener('resize', detectDevTool);
        window.addEventListener('mousemove', detectDevTool);
        window.addEventListener('focus', detectDevTool);
        window.addEventListener('blur', detectDevTool);
    }
}();
```

**Comment by leathakkor** [comment]
@zkte I run this from a cloud server. and that is where i get in to trouble (opening it in the browser). It works fine from my home pc no vpn/proxy. I have a cloud vm that I log in to (tried thee different cloud providers including azure*)  and they all fail the auth check and end up calling "forceKillplayer()"

** Machine Configs:
windows in azure
centos in azure
ubuntu 18 in one non-azure cloud vps
centos 8 in one non-azure cloud vps
ubuntu 22 in another cloud vps

some with x11 forwarding; some with lynx, or just some curl or python. always the auth check gives me a 403 so they must be doing something that blocks cloud providers. or they are doing something else to differentiate my personal machine vs x11 in the cloud on centos. does anyone else run this through a cloud vpn or proxy or anything?

**Comment by zkte** [comment]
@leathakkor looks like blocking vpns, maybe try TOR.
